servlet.py

```diff
@@ -117,9 +118,11 @@ class BaseHandler(tornado.web.RequestHandler):
     max_users_per_pipe = 5
     max_idle_secs = 0
     restart_pipe_after = 1000
+    max_processes = 10
 
     def initialize(self):
         self.callback = self.get_argument('callback', default=None)
+        sem = Semaphore(max_processes)
```

Comment:
This variable needs a more descriptive name, e.g. `doc_pipe_sem`  (we don't just call strings `string`, we call them things like `username`). 

By the way, what is its owner? (How does this work without `self`?)